runtime/src/dispatcher.rs

```diff
@@ -326,126 +473,45 @@ impl Dispatcher {
-                            .expect("add transaction must succeed");
-                    }
-
-                    let (io_write_log, io_root) = txn_tree
-                        .commit(Context::create_child(&ctx))
-                        .expect("io commit must succeed");
-
-                    let header = ComputeResultsHeader {
-                        round: block.header.round + 1,
-                        previous_hash: block.header.encoded_hash(),
-                        io_root: Some(io_root),
-                        state_root: Some(new_state_root),
-                        messages_hash: Some(roothash::Message::messages_hash(&messages)),
-                    };
-
-                    debug!(self.logger, "Transaction batch execution complete";
-                        "previous_hash" => ?header.previous_hash,
-                        "io_root" => ?header.io_root,
-                        "state_root" => ?header.state_root,
-                        "messages_hash" => ?header.messages_hash,
-                    );
-
-                    let rak_sig = if self.rak.public_key().is_some() {
-                        self.rak
-                            .sign(&COMPUTE_RESULTS_HEADER_CONTEXT, &cbor::to_vec(&header))
-                            .unwrap()
-                    } else {
-                        Signature::default()
-                    };
-
-                    let result = ComputedBatch {
-                        header,
-                        io_write_log,
-                        state_write_log,
-                        rak_sig,
-                        messages,
-                    };
+        let results = if check_only {
+            self.txn_check_batch(
+                ctx,
+                id,
+                cache,
+                &protocol,
+                txn_dispatcher,
+                txn_ctx,
+                untrusted_local,
+                inputs,
+                io_root,
+            )
+        } else {
+            self.txn_execute_batch(
+                ctx,
+                id,
+                cache,
+                &protocol,
+                txn_dispatcher,
+                txn_ctx,
+                untrusted_local,
+                inputs,
+                io_root,
+            )
+        };
 
-                    // Send the result back.
-                    protocol
-                        .send_response(id, Body::RuntimeExecuteTxBatchResponse { batch: result })
-                        .unwrap();
-                }
-            }
+        // Return and error response in case of failure.
+        if let Err(error) = results {
+            warn!(self.logger, "Dispatching batch error"; "err" => %error);
+            protocol
+                .send_response(
+                    id,
+                    Body::Error {
+                        module: "".to_owned(), // XXX: Error codes.
+                        code: 0,               // XXX: Error codes.
```

Comment:
Shouldn't at least a different code be used here, so that it doesn't correspond to `CodeNoError`?

Reply:
Good point, so currently this is not used anywhere but makes sense to change to an actual error code :-)